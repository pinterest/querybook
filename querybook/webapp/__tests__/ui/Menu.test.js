import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';

import {
    Menu,
    MenuItem,
    MenuItemPing,
    MenuDivider,
    MenuInfoItem,
} from '../../ui/Menu/Menu';

it('renders without crashing', () => {
    shallow(
        <Menu>
            <MenuItem>
                <MenuItemPing />
            </MenuItem>
            <MenuDivider />
            <MenuInfoItem />
        </Menu>
    );
});

describe('matches enzyme snapshots', () => {
    it('matches snapshot', () => {
        const wrapper = shallow(
            <Menu>
                <MenuItem>
                    <MenuItemPing />
                </MenuItem>
                <MenuDivider />
                <MenuInfoItem />
            </Menu>
        );
        const serialized = toJson(wrapper);
        expect(serialized).toMatchSnapshot();
    });
});

describe('matches test renderer snapshot', () => {
    it('serializes the styles', () => {
        const output = renderer.create(
            <Menu>
                <MenuItem>
                    <MenuItemPing />
                </MenuItem>
                <MenuDivider />
                <MenuInfoItem />
            </Menu>
        );
        expect(output).toMatchSnapshot();
    });
});
