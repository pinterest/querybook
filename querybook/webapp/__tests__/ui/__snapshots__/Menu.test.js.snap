// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches enzyme snapshots matches snapshot 1`] = `
<div
  className="Menu-sc-13z76nv-0 dbaXzY Menu"
>
  <Menu__MenuItem>
    <Menu__MenuItemPing />
  </Menu__MenuItem>
  <Menu__MenuDivider />
  <Menu__MenuInfoItem />
</div>
`;

exports[`matches test renderer snapshot serializes the styles 1`] = `
<div
  className="Menu-sc-13z76nv-0 dbaXzY Menu"
>
  <span
    className="Menu__MenuItem-sc-13z76nv-1 MenuItem"
  >
    <div
      className="Menu__MenuItemPing-sc-13z76nv-4 COjBt MenuItemPing"
    />
  </span>
  <div
    className="Menu__MenuDivider-sc-13z76nv-3 ixcCKO MenuDivider"
  />
  <div
    className="Menu__MenuInfoItem-sc-13z76nv-2 MenuInfoItem"
  />
</div>
`;
